src/main/java/org/apache/commons/text/TokenStringifier.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.text;
+
+/**
+ * Takes a collection of String tokens and combines them into a single String.
+ * <p>
+ * This class functions as the inverse of {@link org.apache.commons.text.StringTokenizer}. All tokens are formatted
+ * by a {@link TokenFormatter} which allows fine grained control over the final output.
+ * </p>
+ */
+public class TokenStringifier {
+
+    /**
+     * The formatter for the delimiter.
+     */
+    private TokenFormatter delimiterFormatter;
+
+    /**
+     * The formatter for the tokens.
+     */
+    private TokenFormatter tokenFormatter;
+
+    /**
+     * Builder used to hold formatted tokens.
+     */
+    private StringBuilder builder;
```

Comment:
Do we need to keep `builder` as a member variable? It seems to be instantiated each time within `stringify()` and isn’t used elsewhere. Would it make sense to keep it local instead?”